test/snapshot/helper/conversions.ts

```diff
@@ -60,37 +71,62 @@ export const sourceToMdapi = async (testDir: string): Promise<string[]> => {
   return dirEntsToPaths(dirEnts);
 };
 
-/** catches missing files by asserting that two directories have the exact same children */
+/**
+ * catches missing files by asserting that two directories have the exact same children
+ * will throw if either directory doesn't exist
+ */
 export const dirsAreIdentical = async (dir1: string, dir2: string): Promise<Chai.Assertion> => {
-  const [files1, files2] = (
-    await Promise.all([
-      fs.promises.readdir(dir1, { recursive: true, withFileTypes: true }),
-      fs.promises.readdir(dir2, { recursive: true, withFileTypes: true }),
-    ])
-  ).map(dirEntsToPaths);
+  const dirs = [dir1, dir2].map(exists);
+
+  const [files1, files2] = (await Promise.all(dirs.map(getAllDirents))).map(dirEntsToPaths).map(resolveRelative(dirs));
+
   return expect(files1).to.deep.equal(files2);
 };
 
+const exists = (dir: string) => {
+  expect(fs.existsSync(dir), `${dir} does not exist`).to.be.true;
+  return dir;
+};
+
+const getAllDirents = (dir: string): Dirent[] => fs.readdirSync(dir, { recursive: true, withFileTypes: true });
+
+const resolveRelative = (parentDirs: string[]) => (subArray: string[], index: number) =>
+  subArray.map(getRelative(parentDirs[index]));
+
+const getRelative = (parent: string) => (child: string) => path.relative(parent, child);
+
+const isFile = (file: fs.Dirent) => file.isFile();
+const getFullPath = (file: fs.Dirent) => path.join(file.path, file.name);
+
 /** dirEnts are sometimes folder, we don't want those.  And we need the full paths */
-const dirEntsToPaths = (dirEnts: fs.Dirent[]): string[] =>
-  dirEnts.filter((file) => file.isFile()).map((file) => path.join(file.path, file.name));
+export const dirEntsToPaths = (dirEnts: fs.Dirent[]): string[] => dirEnts.filter(isFile).map(getFullPath);
 
 const shouldIgnore = (file: string): boolean => {
   // binary zip/unzip isn't exactly the same, so we "skip" that one
   if (file.includes('leafletjs.resource')) return true;
```

Comment:
Should there be another utility for `isBinary()` to detect files like this versus hardcoding specific files?

Reply:
yeah, maybe if this turns out to be pretty common.  

I'm not sure yet if it's "zip/unzip isn't idempotent for **any** binary file" or if it's just "something funny happens with leaflet" because I don't have any other zipped statics in tests yet.

But I like the idea if this is getting out of hand